src/sentry/static/sentry/app/views/settings/organizationMembers/organizationMemberDetail.tsx

```diff
@@ -0,0 +1,421 @@
+import Tooltip from 'app/components/tooltip';
+import recreateRoute from 'app/utils/recreateRoute';
+import space from 'app/styles/space';
+import withOrganization from 'app/utils/withOrganization';
+
+import RoleSelect from './inviteMember/roleSelect';
+
+const MULTIPLE_ORGS = t('Cannot be reset since user is in more than one organization');
+const NOT_ENROLLED = t('Not enrolled in two-factor authentication');
+const NO_PERMISSION = t('You do not have permission to perform this action');
+const TWO_FACTOR_REQUIRED = t(
+  'Cannot be reset since two-factor is required for this organization'
+);
+
+type Props = {
+  organization: Organization;
+} & AsyncView['props'] &
+  RouteComponentProps<
+    {
+      orgId: string;
+      memberId: string;
+    },
+    {}
+  >;
+
+type State = {
+  roleList: Member['roles'];
+  selectedRole: Member['role'];
+  member: Member | null;
+} & AsyncView['state'];
+
+class OrganizationMemberDetail extends AsyncView<Props, State> {
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      roleList: [],
+      selectedRole: '',
+      member: null,
+    };
+  }
+
+  getEndpoints(): [string, string][] {
+    const {organization, params} = this.props;
+    return [
+      ['member', `/organizations/${organization.slug}/members/${params.memberId}/`],
+    ];
+  }
+
+  redirectToMemberPage() {
+    const {location, params, routes} = this.props;
+    const members = recreateRoute('members/', {
+      location,
+      routes,
+      params,
+      stepBack: -2,
+    });
+    browserHistory.push(members);
+  }
+
+  handleSave = async () => {
+    const {organization, params} = this.props;
+
+    addLoadingMessage(t('Saving...'));
+    this.setState({busy: true});
+
+    try {
+      await updateMember(this.api, {
+        orgId: organization.slug,
+        memberId: params.memberId,
+        data: this.state.member,
+      });
+      addSuccessMessage(t('Saved'));
+      this.redirectToMemberPage();
+    } catch (resp) {
+      const errorMessage =
+        (resp && resp.responseJSON && resp.responseJSON.detail) || t('Could not save...');
+      addErrorMessage(errorMessage);
+    }
+    this.setState({busy: false});
+  };
+
+  handleInvite = async (regenerate: boolean) => {
+    const {organization, params} = this.props;
+
+    addLoadingMessage(t('Sending invite...'));
+
+    this.setState({busy: true});
+
+    try {
+      const data = await resendMemberInvite(this.api, {
+        orgId: organization.slug,
+        memberId: params.memberId,
+        regenerate,
+      });
+
+      addSuccessMessage(t('Sent invite!'));
+
+      if (regenerate) {
+        this.setState(state => ({member: {...state.member, ...data}}));
+      }
+    } catch (_err) {
+      addErrorMessage(t('Could not send invite'));
+    }
+
+    this.setState({busy: false});
+  };
+
+  handleAddTeam = (team: Team) => {
+    const {member} = this.state;
+    if (!member!.teams.includes(team.slug)) {
```

Comment:
@billyvg can you remind me what this syntax means again?

Reply:
Ya I ranted about this in #discuss-frontend a bit... `this.state.member` is guaranteed to be not null because we perform the null check in `render()` but I don't know how to make ts recognize this besides creating a new component that expects a non-null member.

Reply:
Any ideas here @EvanPurkhiser ?

Reply:
I think I probably just munge the type typically and type member as always existing for that exact reason.